espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Handlers/Finder.java

```diff
@@ -18,26 +26,51 @@
 
 public class Finder implements RequestHandler {
 
-    public AppiumResponse handle(NanoHTTPD.IHTTPSession session, Map<String, String> uriParams) {
+    public BaseResponse handle(NanoHTTPD.IHTTPSession session, Map<String, String> uriParams)  {
+
+        // If the SessionID is invalid, return InvalidSessionResponse
+        // TODO: Fix SessionID handling redundancies
+        if (!uriParams.get("sessionId").equals(Session.getGlobalSessionId())) {
+            return new InvalidSessionResponse(uriParams.get("sessionId"));
+        }
+
+        // NanoHTTP requires call to parse body before we can get the parameters
+        try {
+            session.parseBody(new HashMap<String, String>());
+        } catch (NanoHTTPD.ResponseException e) {
+            return new BadRequestResponse("Could not parse parameters");
```

Comment:
it might be useful to show the body, which cannot be parsed, in the exception text